app/src/store/playground/playgroundStore.tsx

```diff
@@ -83,7 +87,18 @@ export const createPlaygroundStore = (
   const playgroundStore: StateCreator<PlaygroundState> = (set, get) => ({
     operationType: "chat",
     inputMode: "manual",
-    input: { variables: {} },
+    input: {
+      // TODO(apowell): When implementing variable forms, we should maintain a separate
+      // map of variableName to variableValue. This will allow us to "cache" variable values
```

Comment:
good idea, so values don't get reset if you switch back and forth or accidentally type one extra character or something. Also raises a question for user defined variables in the following flow:
1. add a variable called question_1 to a message
2. set some value to that variable
3. change that variable to be question_2

Don't need to handle this but similar case. Also this will all be a bit different with datasets where there will be a set of potential variables (names + values) coming from the dataset that should be available in the playground

Reply:
Yeah I was wondering about that. We can evict the cache of unused variable values on like prompt run or something like that.

As for the incoming variables from dataset type playground, I will have to play with that to see how it works and where to best parse out the variables.

Reply:
yeah, i will put in some variable parsing logic from spans, so that might inform some things but don't need to worry now